codes/c/chapter_hashing/array_hash_map.c

```diff
@@ -21,11 +21,32 @@ typedef struct {
     int len;
 } MapSet;
 
+/* 用以索引键获取数据 */
+typedef struct {
+    int key[HASH_MAP_DEFAULT_SIZE];
+    int len;
+}IndexMembers;
+
 /* 基于数组简易实现的哈希表 */
 typedef struct {
     Pair *buckets[HASH_MAP_DEFAULT_SIZE];
 } ArrayHashMap;
 
+/* 初始化存储键 */
+void InitIndex(IndexMembers *index){
+    index->len = -1;
+}
+
+/* 将哈希键存入数组 */
+void putKey(IndexMembers *index, int key){
+    index->key[++index->len] = key;
+}
+
+/* 删除哈希键 */
+void DelKey(IndexMembers *index){
+    --index->len;
```

Comment:
How do you guarantte deleting the target key from `*index`

Reply:
Assuming that we call `removeItem(..., key1)` twice, this function will delete two keys(len decreasing by 2), but actually there is just a single target key `key1`.